File: code_snippets/java/src/main/java/develop/ErrorHandling.java
package develop;

import dev.restate.sdk.Context;
import dev.restate.sdk.common.TerminalException;

public class ErrorHandling {

  public void errorHandling(Context ctx) {

    // <start_here>
    throw new TerminalException(500, "Something went wrong");
    // <end_here>

  }
}

File: code_snippets/kotlin/src/main/kotlin/develop/ErrorHandling.kt
package develop

import dev.restate.sdk.common.TerminalException
import dev.restate.sdk.kotlin.Context

class ErrorHandling {
    fun errorHandling(ctx: Context) {

        // <start_here>
        // Inside your handler throw terminal errors with a code and message as follows:
        throw TerminalException(500, "Something went wrong")
        // <end_here>

    }
}
